provider/registry_config/state.go

```diff
@@ -138,6 +138,10 @@ func UpdateRegistryConfigBuilder(ctx context.Context, state *RegistryConfig,
 // PopulateRegistryConfigState takes a Cluster object from CS in input and fills a Terraform state with the data in it.
 // Returns an error if there is a failure
 func PopulateRegistryConfigState(inputCluster *cmv1.Cluster, state *RegistryConfig) error {
+	if state == nil {
```

Comment:
Probably good to move this into the ok check below, in case there is no response from the API you don't really need to init this at all